sdk/runanywhere-kotlin/src/jvmMain/kotlin/com/runanywhere/sdk/services/DeviceRegistrationServiceJvm.kt

```
@@ -0,0 +1,148 @@
+        } catch (e: Exception) {
+            info["cpu_frequency_mhz"] = null
+        }
+
+        // Memory information (in MB)
+        val maxMemory = runtime.maxMemory()
+        val totalMemory = runtime.totalMemory()
+        val freeMemory = runtime.freeMemory()
+        val usedMemory = totalMemory - freeMemory
+        val availableMemory = maxMemory - usedMemory
+
+        info["total_memory_mb"] = maxMemory / (1024 * 1024)
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)